### PR TITLE
MISRA-C general fixes

### PR DESCRIPTION
**Pull request by ceolin**
Assorted MISRA-C fixes for rules:

5.7
5.8
5.9
8.9
15.6

### CODE REVIEW

**Comment by pfl** [comment]
Looks straight forward, +1 from me even though I'm not listed as a reviewer and shippable is still running.

File: doc/kernel/other/polling.rst [hunk 1]
```diff
@@ -172,7 +172,7 @@ to :c:macro:`K_POLL_STATE_NOT_READY` by the user.
         }
     }
 
-Using k_poll_signal()
+Using k_poll_signal_raise()
```
Comment (dbkinder) [inline doc/kernel/other/polling.rst]: The title underline on the next line must be at least as long as the title, so add some more ``===``'s

**Review by dbkinder — DISMISSED** [review]
fix reST to remove doc build warning message

**Review by nashif — APPROVED** [review]

File: doc/kernel/other/polling.rst (hunk 1 above)
Comment (ceolin) [reply doc/kernel/other/polling.rst]: nice catch ! thanks

**Review by jhedberg — APPROVED** [review]
Bluetooth changes look fine

**Review by ioannisg — DISMISSED** [review]
Fix first commit message title: Declarion -> declaration?

**Review by ioannisg — APPROVED** [review]

**Pull request merged**